Refer to this link for build results (access rights to CI server needed): 
http://bock.systems:8080/job/hangups_tox/27/

**Build Log**
_last 25 lines_

```
[...truncated 5 lines...]
 > git config remote.origin.url https://github.com/TheInnerCircleO/hangupsbot # timeout=10
Fetching upstream changes from https://github.com/TheInnerCircleO/hangupsbot
 > git --version # timeout=10
 > git -c core.askpass=true fetch --tags --progress https://github.com/TheInnerCircleO/hangupsbot +refs/pull/*:refs/remotes/origin/pr/*
 > git rev-parse refs/remotes/origin/pr/17/merge^{commit} # timeout=10
 > git rev-parse refs/remotes/origin/origin/pr/17/merge^{commit} # timeout=10
Checking out Revision ff19137b039e2bb2a07faeee548a4fb8f57e2242 (refs/remotes/origin/pr/17/merge)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f ff19137b039e2bb2a07faeee548a4fb8f57e2242
First time build. Skipping changelog.
[workspace] $ /bin/bash /tmp/hudson6099625966166227709.sh
pep8 recreate: /var/lib/jenkins/jobs/hangups_tox/workspace/.tox/pep8
pep8 installdeps: -r/var/lib/jenkins/jobs/hangups_tox/workspace/requirements.txt, -r/var/lib/jenkins/jobs/hangups_tox/workspace/test-requirements.txt
pep8 develop-inst: /var/lib/jenkins/jobs/hangups_tox/workspace
pep8 runtests: PYTHONHASHSEED='1889157099'
pep8 runtests: commands[0] | flake8
./hangupsbot/commands.py:394:80: E501 line too long (84 > 79 characters)
                bot.send_message(event.conv, 'Invalid argument: {}'.format(args[0]))
                                                                               ^
ERROR: InvocationError: '/var/lib/jenkins/jobs/hangups_tox/workspace/.tox/pep8/bin/flake8'
___________________________________ summary ____________________________________
ERROR:   pep8: commands failed
Build step 'Execute shell' marked build as failure
Setting status of 07f0c26c68476dba94df8d83ca3eb9d27c8f540a to FAILURE with url http://bock.systems:8080/job/hangups_tox/27/ and message: Merged build finished.
```

Oh No...

--- a/hangupsbot/commands.py
+++ b/hangupsbot/commands.py
@@ -4,7 +4,6 @@ import json
 import random
 import re
 import requests
-import time
 import datetime
 
 from hangups.ui.utils import get_conv_name

--- a/hangupsbot/commands.py
+++ b/hangupsbot/commands.py
@@ -5,8 +5,7 @@ import random
 import re
 import requests
 import time
-
-from datetime import date as datetime
+import datetime
 
 from hangups.ui.utils import get_conv_name
 from hangupsbot.utils import text_to_segments
@@ -352,22 +351,16 @@ def config(bot, event, cmd=None, *args):
     bot.send_message_segments(event.conv, segments)
 
 
-def _random_date(start, end, format):
-    stime = time.mktime(time.strptime(start, format))
-    etime = time.mktime(time.strptime(end, format))
-    ptime = stime + random.random() * (etime - stime)
-    return time.strftime(format, time.localtime(ptime))
-
-
 @command.register
 def dilbert(bot, event, *args):
     """/bot dilbert [YYYY-MM-DD | random]
     NOTE: First dilbert 1989-04-16"""
 
     dilbert_date_format = '%Y-%m-%d'
-    today = datetime.today().strftime(dilbert_date_format)
 
-    dilbert_modifier = today
+    today = datetime.date.today()
+
+    dilbert_modifier = today.strftime(dilbert_date_format)
 
     segments = [
         hangups.ChatMessageSegment(
@@ -378,26 +371,59 @@ def dilbert(bot, event, *args):
 
     if args:
 
+        first_timestamp = int(
+            datetime.datetime.strptime(
+                '1989-04-16',
+                dilbert_date_format
+            ).strftime('%s')
+        )
+
+        today_timestamp = int(
+            today.strftime('%s')
+        )
+
         if args[0] == 'random':
 
-            dilbert_modifier = _random_date(
-                '1989-04-16',
-                today,
-                dilbert_date_format
-            )
+            random_timestamp = randint(first_timestamp, today_timestamp)
+
+            dilbert_modifier = datetime.date.fromtimestamp(
+                random_timestamp
+            ).strftime(dilbert_date_format)
 
         else:
 
-            valid_arg = re.match('\d{4}-\d{2}-\d{2}', args[0])
+            try:
 
-            if valid_arg is None:
+                timestamp = int(
+                    datetime.datetime.strptime(
+                        args[0],
+                        dilbert_date_format
+                    ).strftime('%s')
+                )
+
+            except:
+
                 bot.send_message(
                     event.conv,
-                    'Invalid argument: {}'.format(args[0])
+                    'Invalid date string: {}'.format(args[0])
                 )
+
                 return
 
-            dilbert_modifier = args[0]
+            date_range = range(first_timestamp, today_timestamp)
+
+            if timestamp not in date_range:
+
+                bot.send_message(
+                    event.conv,
+                    'Date out of range: {}'.format(args[0])
+                )
+
+                return
+
+            dilbert_modifier = datetime.date.fromtimestamp(
+                timestamp
+            ).strftime(dilbert_date_format)
 
     link = 'http://dilbert.com/strip/{}'.format(dilbert_modifier)
 

--- a/hangupsbot/commands.py
+++ b/hangupsbot/commands.py
@@ -391,7 +391,10 @@ def dilbert(bot, event, *args):
             valid_arg = re.match('\d{4}-\d{2}-\d{2}', args[0])
 
             if valid_arg is None:
-                bot.send_message(event.conv, 'Invalid argument: {}'.format(args[0]))
+                bot.send_message(
+                    event.conv,
+                    'Invalid argument: {}'.format(args[0])
+                )
                 return
 
             dilbert_modifier = args[0]

